12주차/17686/src/17686_java_곽유나.java

```diff
@@ -0,0 +1,65 @@
+import java.util.Arrays;
+
+class Solution {
+    String[] answer;
+    public String[] solution(String[] files)
+    {
+        answer=new String[files.length];
+        mergeSort(files,0,files.length-1);
+        return answer;
+    }
```

Comment:
이렇게 하면 아래 mergeSort, checkPriority 메서드를 따로 만들지 않고 유나님의 풀이를 적용할 수 있어보이네요
```suggestion
    public String[] solution(String[] files) {
        answer = new String[files.length];
        Arrays.sort(files, (first, second) -> {
            String delimiter = "\\d";
            String[] firstArr = first.split(delimiter);
            String[] secondArr = second.split(delimiter);
            int priority = firstArr[0].toLowerCase().compareTo(secondArr[0].toLowerCase());
            if (priority != 0) { //first가 순서가 더 빠름
                return priority;
            } //Head가 같을때
            delimiter = "\\D+";
            firstArr = first.split(delimiter);
            secondArr = second.split(delimiter);
            int firstNum = Integer.parseInt(firstArr[1]);
            int secondNum = Integer.parseInt(secondArr[1]);
            return firstNum - secondNum;
        });
        answer = files;
        return answer;
    }
```